test/cleaners/asObject.test.js

```diff
@@ -37,16 +37,28 @@ describe('asObject', function () {
   })
 
   it('accepts shaped objects', function () {
+    // Optional property:
     expect(asDemoObject({ here: 'hi' })).deep.equals({
       here: 'hi',
       maybe: undefined
     })
+
+    // All properties:
     const data = { here: 'hi', maybe: '!', extra: false }
     expect(asDemoObject(data)).deep.equals({
       here: 'hi',
       maybe: '!'
     })
     expect(asDemoObject.withRest(data)).deep.equals(data)
+
+    // Filters __proto__:
+    expect(
+      asDemoObject.withRest(
+        JSON.parse(
+          '{ "here": "hi", "maybe": "!", "extra": false, "__proto__": false }'
+        )
+      )
+    ).deep.equals(data)
```

Comment:
Just curious, when does `Object.keys` include `"__proto__"` in the returned array?

Reply:
Try plugging this into node, and behold the madness: `Object.keys(JSON.parse('{ "__proto__": false }'))`